builtin/providers/aws/provider.go

```diff
@@ -145,6 +145,13 @@ func Provider() terraform.ResourceProvider {
 					return hashcode.String(v.(string))
 				},
 			},
+
+			"dynamodb_endpoint": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
```

Comment:
@phinze that's correct based on their docs: https://github.com/aws/aws-sdk-go/blob/67dc9f948602be9f85cb640f89b0adec994ccbda/aws/config.go#L49-L55 and I've also confirmed that locally.

But I can remove the `Default: ""` above and make it more explicit:

```go
awsDynamoDBConfig := &aws.Config{
	Credentials: creds,
	Region:      c.Region,
	MaxRetries:  c.MaxRetries,
}

if c.DynamoDBEndpoint != nil {
	awsDynamoDBConfig.Endpoint = c.DynamoDBEndpoint
}
```